### ISSUE TITLE
DXT Dimension Restrictions

### ISSUE THREAD

**Issue by SirWaddles** [image-rs/image#872]
One of the requirements for the `DXTImageDecoder` is that the image for it have a width and height that are both powers of two.

Admittedly, I'm not all that familiar with the internals of the DXT image format, but as far as I can tell (and please correct me if I'm wrong), the only technical restriction is that the dimensions be a multiple of four. 

For another program, I was using the [libsquish library](https://github.com/Cavewhere/squish) to do DXT decompression, and it has no such restriction.

If it's okay with everyone here, I'd like to make a pull request removing the restriction (if it's possible, anyway).

Thanks

**Comment by CensoredUsername** [comment]
I can't remember why I put that check in there when I wrote the base DXT implementation. IIRC .dds files are limited to power-of-two widths and heights for DXT formats but the format indeed doesn't have any restrictions except for requiring multiples of 4x4 pixel chunks.

Feel free to remove it.

**Comment by jansol** [comment]
May I suggest [my libsquish port](https://github.com/jansol/squish-rs) in place of said code? It's a pretty 1:1 RIIR of the original code, handles NPOT images and implements all 3 color compression schemes present in the original lib. /shameless plug

FWIW I've cleaned up the API a bit since the last time I uploaded it to crates.io. Will upload a new version once I get the documentation up to speed as well.

**Comment by CensoredUsername** [comment]
@jansol interesting code. When I wrote the DXT implementation for image there were still legal concerns around DXT compression licensing, so it's a simple clean room implementation.

Due to this the encoder is a bit simpler than what could be (it's a brute force search with some special cases). Originally I did consider principle component analysis (looking at your code that's one of the strategies it uses) but due to it requiring significant amounts of floating point math which would in the end be about as fast as just brute forcing it I went for the simpler approach.

I wonder which implementation is faster / more accurate. The core brute force search I used should scale pretty well as it's completely operating in fixed point and the inner loops get optimized to SIMD quite well. Might be fun to benchmark.
